File: lib/add_admin_role.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:firebase_auth/firebase_auth.dart';
import 'package:firebase_core/firebase_core.dart';

void main() async {
  // Initialize Firebase
  await Firebase.initializeApp();
  
  final user = FirebaseAuth.instance.currentUser;
  if (user == null) {
    print('❌ No user logged in');
    return;
  }
  
  print('✅ Current user: ${user.email} (${user.uid})');
  
  // Add admin role
  await FirebaseFirestore.instance
      .collection('users')
      .doc(user.uid)
      .set({
    'roles': FieldValue.arrayUnion(['admin'])
  }, SetOptions(merge: true));
  
  print('✅ Admin role added successfully!');
  print('👉 Refresh your browser and try sending notifications again.');
}

File: functions/add_admin_role.js
const admin = require('firebase-admin');

// Initialize Firebase Admin
admin.initializeApp({
  projectId: 'truehome-9a244',
});

const db = admin.firestore();

async function addAdminRole() {
  try {
    // Get user by email
    const email = 'truehome376@gmail.com';
    
    console.log(`Searching for user with email: ${email}`);
    
    const usersSnapshot = await db.collection('users')
      .where('email', '==', email)
      .get();
    
    if (usersSnapshot.empty) {
      console.log('No user found with that email');
      return;
    }
    
    const userDoc = usersSnapshot.docs[0];
    console.log(`Found user: ${userDoc.id}`);
    
    // Update user with admin role
    await userDoc.ref.update({
      roles: admin.firestore.FieldValue.arrayUnion('admin'),
      role: 'admin',
    });
    
    console.log('✅ Admin role added successfully!');
    
    // Verify the update
    const updatedDoc = await userDoc.ref.get();
    console.log('Updated user data:', updatedDoc.data());
    
    process.exit(0);
  } catch (error) {
    console.error('❌ Error:', error);
    process.exit(1);
  }
}

addAdminRole();
